### ISSUE TITLE
[lxqt-globalkeys][lxqt-panel] key grabbing

### ISSUE THREAD

**Issue by vicencb** [lxde/lxqt#618]
Hello,
this issue is to report two related bugs and an enhancement.
1) First bug:
When lxqt-globalkeys has a key combination disabled it's still grabbing it.
By default Ctrl+F2 is set to switch to the second desktop. This shortcut can not be removed, neither in the GUI or the configuration file. As a workaround there is the option to disable it.
When disabled it stops switching desktops, but aplications don't receive this because lxqt-globalkeys still grabs it.
To fix this, can the shortcuts be removable or, at least, un-grab disabled shortcuts?

2) Second bug:
XF86Audio* keys are grabbed by openbox (I've configured it that way).
Under this situation lxqt-globalkeys fails to grab them and shows a pop-up window to indicate the problem.
At this point the "Volume control" widget fails to start due to the previous error.
To fix this, please, read the enhancement.

3) The enhancement:
The "Volume control" and the "Show desktop" widgets have both a hard dependency on lxqt-globalkeys. They fail to start when lxqt-globalkeys is not present or fails to register the associated shortcuts.
To improve this, can lxqt-globalkeys be an option instead of a requirement? This involves at least the two mentioned widgets, I don't know if there are other components requiring it.

All of this has been tested with lxqt version 0.9.0.

Thanks in advance!
Regards,
  Vicenç.

**Comment by flamusdiu** [comment]
@vicencb umm, is that why sometimes the XF86Audio* keys are sometimes not grabbed by the Volume Control widget? Also, what benefits do you get using OpenBox for the keys instead of the widget?

**Comment by vicencb** [comment]
I don't understand the first question, can you rephrase it? Please.
Regarding the second one:
with OpenBox there is already a keyboard shortcut manager which is not optional, it is required for window switching and other window related operations.
With one shortcut manager already in place, any other one is optional, so lxqt-globalkeys can be removed.
The benefits on removing lxqt-globalkeys is that it won't get on my way [of grabbing key combinations targetted to applications].

Regards,
  Vicenç.

**Comment by flamusdiu** [comment]
@vicencb sometimes the Widget doesn't grab the keys (maybe it's a Linux 4.0 issue as well) and I can't use the XF86Audio* keys to control the widget.

Anyway, are you still using the 'Volume Control' widget or something else?

**Comment by vicencb** [comment]
I don't know, but wouldn't expect any widgets grabbing keys directly, because if they did, which was the purpose of lxqt-globalkeys?
And yes, I'm still wanting to use the "Volume control" widget as an status monitor of the current volume settings and also to control it with the mouse.
Basically, point 3 in the issue report would fix everything. Just modifying the widgets that make use of lxqt-globalkeys to carry on instead of bailing out when lxqt-globalkeys is not available.

**Comment by flamusdiu** [comment]
gotcha. I get errors when logging in where lxqt-panel can't set the XF86Audio* keys. Really annoying. =\

**Comment by Vladimir-csp** [comment]
I agree that failing to register a keybinding or absence of lxqt-globalkeys should not be treated as critical error for any component. There should be a soft dependency.
And IMHO, lxqt-globalkeys would serve more logically as a centralized tool for configuring shortcuts for components, rather than a thing that actually grabs and listens for keys itself.

**Comment by palinek** [comment]
> And IMHO, lxqt-globalkeys would serve more logically as a centralized tool for configuring shortcuts for components, rather than a thing that actually grabs and listens for keys itself.

Isn't it? You can configure/disable all keybindings that are requested by panel components in `lxqt-config-globalkeyshortcuts`:
![screenshot](https://cloud.githubusercontent.com/assets/10990929/7342427/3676e23e-ecb4-11e4-9693-9c5ccb9eada7.png)
![screenshot](https://cloud.githubusercontent.com/assets/10990929/7342430/48e98e62-ecb4-11e4-9289-0e899fbb229c.png)

**Comment by Vladimir-csp** [comment]
@palinek I know that. I meant it does both, but actual grabbing and listening should go to the components. Or is it the case already and i`m missing the whole deal?

**Comment by palinek** [comment]
I don't know the rationale of using the daemon also for grabbing keys but I suppose ther is some ... @luis-pereira, @jleclanche ?

@Vladimir-csp what would be the benefit of removing grabbing keys from a centralized daemon? The daemon has to run anyway for components to get the correct keybindings for actions (that the user could have changed)...

**Comment by Vladimir-csp** [comment]
Is the daemon really needed? The components could just read and watch the config, lxqt-globalkeys would be just a config editor. One daemon less to run.

**Comment by kuzmas** [comment]
It's much easier to grab keys in one place. Grabbing is really X11-specific and is performed in special thread. Copying this functionality to any widget/plugin/component which needs key binding is not wise at all IMO.

**Comment by kuzmas** [comment]
What is a good fix (as I can see it) - is when a shortcut is disabled it has to be ungrabbed from X11. (as @vicencb  suggested https://github.com/lxde/lxqt/issues/618#issue-70797310)

**Comment by kuzmas** [comment]
I forgot to mention: daemon is also needed to call key-bound commands and dbus calls.
